edgelet/hsm-rs/src/crypto.rs

```diff
@@ -194,6 +194,30 @@ fn make_certification_props(props: &CertificateProperties) -> Result<CERT_PROPS_
             unsafe { cert_properties_destroy(handle) };
             ErrorKind::CertProps
         })?;
+
+    if props.san_entries.len() > 0 {
+        let result: Result<Vec<CString>, NulError> = props
+            .san_entries
+            .iter()
+            .map(|s: &String| CString::new(s.clone()))
+            .collect();
+
+        let result: Vec<CString> = result.ok().ok_or_else(|| {
+            unsafe { cert_properties_destroy(handle) };
+            ErrorKind::CertProps
+        })?;
+
+        let result: Vec<*const c_char> = result.iter().map(|s| s.as_ptr()).collect();
```

Comment:
In general, I like to post-pone calling `collect` till the very end and just deal with iterators in the interim to avoid allocs.

Reply:
Not relevant per our convo.